### DISCUSSION TITLE
Changing IP of a VM

### DISCUSSION THREAD

**Discussion by kieron** [cockpit-project/cockpit-machines#1149] [General]
I'm going to be running my ApisCP web server in a virtual machine in Cockpit, so I bought an extra IP on OVH and assigned it to my dedicated server which is what I'm running Cockpit on.

But what I haven't yet figured out is how to give the virtual machine that IP, I feel like I've looked everywhere, so I'm wondering if that's not available from the cockpit machines web UI?

The virtual machine is running rocky 8.8 if that helps.

So yeah just looking for clarification on the above and if anyone has any tips I'd appreciate it.

Thanks

**Comment by skobyda** [comment]
Hey, this is currently not avaialble in our UI.
In our UI, you can setup virtual network's DHCP to give out addresses from a certain range, so maybe just limiting the range only to 1 IP address could work? 
![Screenshot from 2023-07-14 08-59-44](https://github.com/cockpit-project/cockpit-machines/assets/42733240/8da1e335-4e46-4639-88e5-b83259ed62de)

But more straighforward solution can be achieved through CLI. Run command `virsh edit [vmName]`, and then edit network interface by adding IP element there, as shown here in libvirt documentation: https://libvirt.org/formatdomain.html#ip-configuration

So then in the XML, network interface which you want an IP address assign to would have configuration like: `<ip address='192.168.122.5' prefix='24'/>`

**Reply by kieron** [reply]
Hey, I can try your first suggestion - basically I just need the VM to be open to the web like a web server should be, I've got a domain I need to point to it also. 

I bought a second IP because I assumed that would be necessary so thought I'd need to give it to the VM

Thank you
